### PR TITLE
C1 khloud remove words from project

### PR DESCRIPTION
**Pull request by Khloud-Khalid**
[Issue #183]

UML reviewer: @nourhanZ 
code reviewer: @adamggg 
documentation reviewer: @zoheiry 
scenario reviewer: @KareemAliRahman 
test reviewer: @zoheiry

files changed concerning story: 
controllers/projects_controller.rb
views/projects/show.html.rb

### CODE REVIEW

File: arability/app/assets/javascripts/game.js [hunk 1]
```diff
@@ -180,6 +180,8 @@ function dropAblockCont(clss, btn, randNum, counter){
```
Comment (adamggg) [inline arability/app/assets/javascripts/game.js:179]: Why is game.js here?

File: arability/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -12,55 +11,511 @@ class ProjectsController < ApplicationController
   #   on success: returns an array of projects of the developer currently logged in.
   #   on failure: notifies the user that he can't see this page.
   def index
- 	  developer = Developer.where(:gamer_id => current_gamer.id).first
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb:38]: Why did you delete Timo's code?

File: arability/app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -12,55 +11,511 @@ class ProjectsController < ApplicationController
...
-    @project = Project.createproject(params[:project])
-    respond_to do |format|
-      if @project.save
-        format.html { redirect_to "/projects", notice: 'Project was successfully created.' }
-        format.json { render json: @project, status: :created, location: @project }
+#     object not valid (no project or word id), word already exists in project, keyword or synonym does not exist, 
+#     developer trying to add word is not owner of the project nor is the project shared with him/her, not registered
+#     developer, or keyword is not in the project.
+def add_word
+    # if the word doesn't have synonyms redirect to follow word
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Remove this if you just commented and don't need it

File: arability/app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -12,55 +11,511 @@ class ProjectsController < ApplicationController
...
-    respond_to do |format|
-      format.html
-      format.json { render json: @project }
+#   keyword does not exist or is not in the project, not registered developer.
+  def remove_word
+    if Developer.find_by_gamer_id(current_gamer.id) != nil 
+      @project_id = params[:project_id]
+      @word_id = params[:word_id]
+      @removed_word = PreferedSynonym.where(keyword_id: @word_id).all
+      @removed_word.each { |word|
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: This is not the convention that we follow use ar.each do |a|

File: arability/app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -12,55 +11,511 @@ class ProjectsController < ApplicationController
...
+    if Developer.find_by_gamer_id(current_gamer.id) != nil 
+      @project_id = params[:project_id]
+      @word_id = params[:word_id]
+      @removed_word = PreferedSynonym.where(keyword_id: @word_id).all
+      @removed_word.each { |word| 
+        if word.project_id = @project_id
+          @remove = word
+        end 
+      }
+      if  @remove != nil
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb:495]: why 2 spaces before the @?

File: arability/app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -12,55 +11,511 @@ class ProjectsController < ApplicationController
...
+        @remove.destroy
+        flash[:success] = t(:word_removed_successfully)
+        redirect_to project_path(@project_id), flash: flash
+        return
+      else
+        flash[:notice] = t(:word_does_not_exist)
+        redirect_to project_path(@project_id), flash: flash
+        return
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Why do you need to return here?

File: arability/app/views/projects/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+					<button class="btn"><i class="icon-share"></i> <%= t(:share) %></button>
+				</div>
+				<div class="btn-group">
+					<button class="btn"><i class="icon-edit"></i> <%= t(:edit_project) %> <span></span></button>
+				</div>
+				<div class="btn-group">
+					<button class="btn"><i class="icon-trash"></i> <%= t(:delete) %> <span></span></button>
+				</div>
+				<div class="btn-group">
+					<button class="btn"><i class="icon-plus-sign"></i> <%= link_to t(:Add_words) , {action: "search", controller: "search", :project_id => @project.id}, {:style=>'color:#000000', :class => "css_class"} %> <span></span></button>
```
Comment (adamggg) [inline arability/app/views/projects/show.html.erb]: Why the empty span at the end?

File: arability/app/assets/javascripts/game.js (hunk 1 above)
Comment (Khloud-Khalid) [inline arability/app/assets/javascripts/game.js:179]: it got here when i pulled master

File: arability/app/controllers/projects_controller.rb (hunk 2 above)
Comment (Khloud-Khalid) [inline arability/app/controllers/projects_controller.rb:38]: Timo was the one who modified it, i just pulled his branch.

File: arability/app/controllers/projects_controller.rb (hunk 3 above)
Comment (Khloud-Khalid) [inline arability/app/controllers/projects_controller.rb]: this method is from another story, it's already modified in its pull request.
